plume-models/src/timeline/mod.rs

```diff
@@ -273,6 +272,70 @@ mod tests {
             assert_eq!(1, tl_u2.len());
             assert_eq!(new_tl1_u2, tl_u2[0]);
 
+            Ok(())
+        });
+    }
+
+    #[test]
+    fn test_timeline_creation_error() {
+        let conn = &db();
+        conn.test_transaction::<_, (), _>(|| {
+            let users = userTests::fill_database(conn);
+
+            assert!(Timeline::new_for_user(
+                conn,
+                users[0].id,
+                "my timeline".to_owned(),
+                "invalid keyword".to_owned(),
+            )
+            .is_err());
+            assert!(Timeline::new_for_instance(
+                conn,
+                "my timeline".to_owned(),
+                "invalid keyword".to_owned(),
+            )
+            .is_err());
+
+            assert!(Timeline::new_for_user(
+                conn,
+                users[0].id,
+                "my timeline".to_owned(),
+                "author in non_existant_list".to_owned(),
+            )
+            .is_err());
+            assert!(Timeline::new_for_instance(
+                conn,
+                "my timeline".to_owned(),
+                "lang in dont-exist".to_owned(),
+            )
+            .is_err());
+
+            List::new(conn,
+                      "friends",
+                      Some(&users[0]),
+                      ListType::User
+                      ).unwrap();
+            List::new(conn,
+                      "idk",
```

Comment:
haah

Reply:
I'm not good at this whole "giving name to lists" thing

Reply:
i disagree, i think you're very good 😸